src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -99,111 +97,102 @@ abstract class Particle : WasmObject() {
 }
 
 abstract class Handle : WasmObject() {
-    var name: String? = null
-    var particle: Particle? = null
+    lateinit var name: String
+    lateinit var particle: Particle
     abstract fun sync(encoded: String)
     abstract fun update(added: String, removed: String)
 }
 
-open class Singleton<T : Entity<T>> constructor(val entityCtor: () -> T) : Handle() {
+open class Singleton<T : Entity<T>>(val entityCtor: () -> T) : Handle() {
     private var entity: T? = null
 
     override fun sync(encoded: String) {
-        entity = entityCtor.invoke().decodeEntity(encoded)
+        entity = entityCtor().decodeEntity(encoded)
     }
 
     override fun update(added: String, removed: String) {
-        entity = entityCtor.invoke().decodeEntity(added)
+        entity = entityCtor().decodeEntity(added)
     }
 
-    fun get(): T? {
-        return entity
-    }
+    fun get() = entity
 
     fun set(entity: T) {
         this.entity = entity
         val encoded = entity.encodeEntity()
         singletonSet(
-            this.particle!!.toWasmAddress(),
-            this.toWasmAddress(),
+            particle.toWasmAddress(),
+            toWasmAddress(),
             encoded.toWasmString()
         )
     }
 
     fun clear() {
         entity = entityCtor()
-        singletonClear(this.particle!!.toWasmAddress(), this.toWasmAddress())
+        singletonClear(particle.toWasmAddress(), toWasmAddress())
     }
 }
 
-class Collection<T : Entity<T>> constructor(private val entityCtor: () -> T) : Handle(),
+class Collection<T : Entity<T>>(private val entityCtor: () -> T) : Handle(),
     Iterable<T> {
-    override fun iterator(): Iterator<T> {
-        return entities.values.iterator()
-    }
 
     private val entities: MutableMap<String, T> = mutableMapOf()
 
+    val size: Int
+        get() = entities.size
+
+    override fun iterator() = entities.values.iterator()
+
     override fun sync(encoded: String) {
         entities.clear()
         add(encoded)
     }
 
     override fun update(added: String, removed: String) {
         add(added)
-        val decoder = StringDecoder(removed)
-            var num = decoder.getInt(":")
+        with(StringDecoder(removed)) {
+            var num = getInt(":")
             while (num-- > 0) {
-                val len = decoder.getInt(":")
-                val chunk = decoder.chomp(len)
+                val len = getInt(":")
+                val chunk = chomp(len)
                 // TODO: just get the id, no need to decode the full entity
-                val entity = entityCtor.invoke().decodeEntity(chunk)!!
+                val entity = entityCtor().decodeEntity(chunk)!!
                 entities.remove(entity.internalId)
             }
+        }
     }
 
-    fun size(): Int {
-        return entities.size
-    }
-
-    fun empty(): Boolean {
-        return entities.isEmpty()
-    }
+    fun empty() = entities.isEmpty()
```

Comment:
`isEmpty` would probably be a better name for this, but it can wait until later like with the other API-changing issues

Reply:
Done :)